### PR TITLE
feat: Adding random state fixing to parameter experiments to ensure comparability

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

Generate a set of random states (one per 'run') for scenario testing to ensure that for each parameter setting a broadly comparable chain of events is observed.

### CODE REVIEW

File: vega_sim/scenario/ideal_market_maker_v2/scenario.py [hunk 1]
```diff
@@ -145,6 +151,7 @@ def set_up_background_market(
             tag=str(tag),
             buy_intensity=self.buy_intensity,
             sell_intensity=self.sell_intensity,
+            random_state=random_state,
```
Comment (Galen-Cao) [inline vega_sim/scenario/ideal_market_maker_v2/scenario.py:154]: Should we also add 'random state' in MarketOrderTrader agent when posting MOs to fix randomness?

**Review by Galen-Cao — COMMENTED** [review]
Leave a comment, please ignore if doesn't make sense

File: vega_sim/scenario/ideal_market_maker_v2/scenario.py (hunk 1 above)
Comment (TomMcL) [reply vega_sim/scenario/ideal_market_maker_v2/scenario.py:154]: I think that's what this is doing here no? Or am I missing using it somewhere?

Comment (Galen-Cao) [reply vega_sim/scenario/ideal_market_maker_v2/scenario.py:154]: Ah, I confused it with another agent, my bad. Looks fine to me now.

**Pull request merged**